addons/udes_advanced_packaging/models/stock_quant_package.py

```diff
@@ -101,3 +101,36 @@ def write(self, vals):
         """
         self.validate_parent_name(vals)
         return super().write(vals)
+
+    def _get_all_products_quantities(self):
+        """This function computes the different product quantities for the given package"""
+        # TDE CLEANME: probably to move somewhere else, like in pack op
+        res = {}
+        for quant in self._get_contained_quants():
+            if quant.product_id not in res:
+                res[quant.product_id] = 0
+            res[quant.product_id] += quant.quantity
+        return res
+
+    def assert_reserved_full_package(self, move_lines):
+        """Check that a package is fully reserved at move_lines."""
+        MoveLine = self.env["stock.move.line"]
+
+        self.ensure_one()
+
+        pack_products = frozenset(self._get_all_products_quantities().items())
+        mls_products = frozenset(move_lines._get_all_products_quantities().items())
+        if pack_products != mls_products:
+            # move_lines do not match the quants
+            picking = move_lines.mapped("picking_id")
+            picking.ensure_one()
+            pack_mls = MoveLine.search(
```

Comment:
It is the second time we have the same search, we should put this into a function: `_get_current_move_lines()`, probably we want it in `udes_stock`, since seems a useful helper function.